src/main/java/com/uber/cadence/worker/WorkerOptions.java

```diff
@@ -17,112 +17,140 @@
+
+        /**
+         * When set to true doesn't poll on workflow task list even if there are registered workflows with a worker.
+         * For clarity prefer not registing workflow types with a {@link Worker} to setting this option.
+         * But it can be useful for disabling polling through configuration without a code change.
+         */
+        public Builder setDisableWorkflowWorker(boolean disableWorkflowWorker) {
+            this.disableWorkflowWorker = disableWorkflowWorker;
+            return this;
+        }
+
+        /**
+         * When set to true doesn't poll on activity task list even if there are registered activities with a worker.
+         * For clarity prefer not registing activity implementations with a {@link Worker} to setting this option.
+         * But it can be useful for disabling polling through configuration without a code change.
+         */
+        public Builder setDisableActivityWorker(boolean disableActivityWorker) {
+            this.disableActivityWorker = disableActivityWorker;
+            return this;
+        }
+
+        /**
+         * Override human readable identity of the worker. Identity is used to identify a worker and is recorded in
+         * the workflow history events.
+         * For example when a worker gets an activity task the correspondent ActivityTaskStarted event contains
+         * the worker identity as a field.
+         * Default is whatever <code>(ManagementFactory.getRuntimeMXBean().getName()</code> returns.
+         */
+        public Builder setIdentity(String identity) {
+            this.identity = identity;
+            return this;
+        }
+
+        /**
+         * Override a data converter implementation used by workflows and activities executed by this worker.
+         * Default is {@link com.uber.cadence.converter.JsonDataConverter} data converter.
+         */
+        public Builder setDataConverter(DataConverter dataConverter) {
+            if (dataConverter == null) {
+                throw new IllegalArgumentException("null");
+            }
+            this.dataConverter = dataConverter;
+            return this;
+        }
+
+        /**
+         * Maximum number of activities started per second.
+         * Default is 0 which means unlimited.
+         */
+        public Builder setWorkerActivitiesPerSecond(double workerActivitiesPerSecond) {
+            this.workerActivitiesPerSecond = workerActivitiesPerSecond;
+            return this;
+        }
+
+        /**
+         * Maximum number of parallely executed activities.
+         */
+        public Builder setMaxConcurrentActivityExecutionSize(int maxConcurrentActivityExecutionSize) {
+            this.maxConcurrentActivityExecutionSize = maxConcurrentActivityExecutionSize;
+            return this;
+        }
+
+        /**
+         * Maximum size of a thread pool used to execute workflows.
+         */
+        public Builder setMaxWorkflowThreads(int maxWorkflowThreads) {
```

Comment:
why expose MaxWorkflowThreads as a config option for Workflows but MaxConcurrentActivityExecutionSize for activities instead ?